snuba/clickhouse/http.py

```diff
@@ -31,20 +28,45 @@ def __repr__(self) -> str:
 
 
 class HTTPBatchWriter(BatchWriter):
-    def __init__(self, schema, host, port, options=None, table_name=None):
-        self.__schema = schema
+    def __init__(self,
+        schema,
+        host,
+        port,
+        encoder: Callable[[WriterTableRow], bytes],
+        options=None,
+        table_name=None,
+        chunk_size: int = 1,
+    ):
+        """
+        Builds a writer to send a batch to Clickhouse.
+
+        :param schema: The dataset schema to take the table name from
+        :param host: Clickhosue host
+        :param port: Clickhosue port
+        :param encoder: A function that will be applied to each row to turn it into bytes
+        :param options: options passed to Clickhouse
+        :param table_name: Overrides the table coming from the schema (generally used for uplaoding
+            on temporary tables)
+        :param chunk_size: The chunk size (in rows).
+            We send data to the server with Transfer-Encoding: chunked. If 0 we send the entire
+            content in one chunk.
+        """
         self.__pool = HTTPConnectionPool(host, port)
         self.__options = options if options is not None else {}
         self.__table_name = table_name or schema.get_table_name()
+        self.__chunk_size = chunk_size
+        self.__encoder = encoder
 
-    def __default(self, value):
-        if isinstance(value, datetime):
-            return value.strftime(DATETIME_FORMAT)
-        else:
-            raise TypeError
+    def _prepare_chunks(self, rows: Iterable[WriterTableRow]) -> Generator[bytes, None, None]:
```

Comment:
I think you should be able to leave this as `-> Iterable[bytes]` when you're using a generator that only yields.